autoextract_poet/items.py

```diff
@@ -170,45 +234,324 @@ class ProductFromList(Item):
 
     @classmethod
     def from_dict(cls, item: Optional[Dict]):
+        new_item = _apply_types(
+            item,
+            from_dict={'aggregateRating': Rating},
+            from_list={'offers': Offer}
+        )
+        return super().from_dict(new_item)
+
+
+@attr.s(auto_attribs=True, slots=True)
+class ProductList(Item):
+
+    url: Optional[str] = None
+    products: List[ProductFromList] = attr.Factory(list)
+    breadcrumbs: List[Breadcrumb] = attr.Factory(list)
+    paginationNext: Optional[PaginationLink] = None
+    paginationPrevious: Optional[PaginationLink] = None
+
+    @classmethod
+    def from_dict(cls, item: Optional[Dict]):
+        # XXX: why is None not preserved for pagination?
```

Comment:
Actually, `None` is preserved for pagination. Maybe the use of `{}` in the get lead to confusion. I've updated the code to use the new procedure here: https://github.com/scrapinghub/autoextract-poet/pull/14/commits/edd90b1885c290de09231cc99039752484f1148a

Reply:
Thanks @ivanprado! `{}` is what got me confused indeed. In https://github.com/scrapinghub/autoextract-poet/commit/edd90b1885c290de09231cc99039752484f1148a you updated it for article lists; would you mind fixing it here (for product lists), and for reviews as well?

Reply:
Sure, done here: https://github.com/scrapinghub/autoextract-poet/pull/14/commits/d9972172906c1626f9e6e881c00da0a2ed8cbc11